### ISSUE TITLE
Trying to access array offset on value of type bool

### ISSUE THREAD

**Issue by SkywalkerLajos** [sabre-io/dav#1531]
Hello, I have a laravel 9.52.4, php 8.2, mysql 8.0.31 project which is a calendar system and I would like to synchronize the Thunderbird's calendar with my own. So if someone create an event in the Thunderbird's calendar then that event should appear in my own, i mean this event should be stored in the database of my calendar. In addtition, if someone crates an event in my calendar then that event should appear in the Thunderbird's calendar. As far as i know, the sabre-io/dav can do this.

I followed the steps according to the documentation:
- i ran in the laravel project folder (www/laravel/): composer require sabre/dav ~3.2.0, composer update sabre/dav
- then: mkdir data, mkdir public, chmod a+rwx data public
- my project folder contained already a server.php file, so i copied to the beginning of the file : 
...
$rootDirectory = new DAV\FS\Directory('public');
// The server object is responsible for making sense out of the WebDAV protocol
$server = new DAV\Server($rootDirectory);
...
- then: cat examples/sql/mysql.* | mysql -u root -p sabredav -h 127.0.0.1
The database built successfully.
-  i copied the calendarserver.php file to my project (www/laravel/calendarserver.php). 
- $pdo = new PDO('mysql:dbname=sabredav;host=127.0.0.1', 'root', 'password');

Then i want to test it: localhost/.../laravel/calendarserver.php/
but some error occured: 
Optional parameter $realm declared before required parameter $response ...
Argument #1 ($request) must be of type Sabre\HTTP\RequestInterface, string given, called in
preg_match_all(): Passing null to parameter #2 ($subject)
etc.

I fixed them, after all this error occured:
This XML file does not appear to have any style information associated with it. The document tree is shown below.
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
<s:sabredav-version>3.2.2</s:sabredav-version>
<s:exception>ErrorException</s:exception>
<s:message>Trying to access array offset on value of type bool</s:message>
</d:error>

I dont know how to fix, where to find the source of the error.

Did I do something wrong during installation?
Can anyone give me some advice?

Thanks in advance!

**Comment by phil-davis** [comment]
I see that Laravel major version 9 is for PHP 8.0 to 8.2
https://laravel.com/docs/9.x/releases

`sabre/dav` support for PHP 8 is in the `sabre/dav` major version 4.
It started from https://github.com/sabre-io/dav/releases/tag/4.1.2

I suggest that you use the latest `sabre/dav` https://github.com/sabre-io/dav/releases/tag/4.6.0 - that should work much better with PHP 8, which checks function parameter and return types.

**Comment by SkywalkerLajos** [comment]
I apologize, but I recently graduated and have limited professional experience in this field. This is one of my first tasks.

So I updated the sabre/dav package to v. 4.1.2

Then:
mkdir data
mkdir public
chmod a+rwx data public

Finally, i should create a file called server.php, and enter the given code from the Getting Started page.

My question is:
My laravel project contains a server.php file by default, which is necessary for the project to work.
server.php code:
$uri = urldecode(
    parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH)
);

if ($uri !== '/' && file_exists(__DIR__.'/public'.$uri)) {
    return false;
}

require_once __DIR__.'/public/index.php';

Should I include the code in this file from the Getting Started page with the correct URL? Below this code, or above? 

What is the next step to synchronize the Thunderbird's calendar with my own laravel calendar system?

Thanks in advance!
